src/services/goalTemplates.ts

```diff
@@ -229,15 +229,10 @@ export async function getFieldPromptsForCuratedTemplate(
       }
       return promptsWithResponses;
     },
-    // the inital set of prompts, including the
-    // response key if not already present
-    prompts.map((p: FieldPrompts) => {
-      if (p.response) {
-        return p;
-      }
-
-      return { ...p, response: [] };
-    }),
+    // the inital set of prompts, which can't have a response yet
+    // because `prompts` is an array of GoalTemplateFieldPromptModel, which both
+    // doesn't include a response and doesn't join with GoalFieldResponseModel.
+    prompts.map((p: FieldPrompts) => ({ ...p, response: [] })),
```

Comment:
I'm pretty confident after looking through the code that this branch:

```js
if (p.response) {
  return p;
}
```

...is unused in all cases and can therefore be removed (comment updated with reasoning)